Classes/common/HTTP/CDTURLSession.m

```diff
@@ -53,8 +62,29 @@ - (CDTURLSessionTask *)dataTaskWithRequest:(NSURLRequest *)request
                                                         NSURLResponse *response,
                                                         NSError *error))completionHandler
 {
+    
+    //make the context
+    CDTHTTPInterceptorContext *ctx = [[CDTHTTPInterceptorContext alloc]
+                                            initWithRequest:[request mutableCopy]];
+    for (NSObject *obj in self.requestInterceptors) {
+        if([obj conformsToProtocol:@protocol(CDTHTTPInterceptor)]) {
+            NSObject<CDTHTTPInterceptor> *interceptor =
+                (NSObject<CDTHTTPInterceptor> *)obj;
+
+            if([interceptor respondsToSelector:@selector(interceptRequestInContext:)]) {
+                ctx = [interceptor interceptRequestInContext:ctx];
+            } else {
+                CDTLogWarn(CDTREPLICATION_LOG_CONTEXT, @"%@ doesn't respond to \
+                           interceptRequestInContext:, skipping",interceptor);
+            }
+        } else {
+            CDTLogWarn(CDTREPLICATION_LOG_CONTEXT, @"%@ doesn't conform to protocol \
+                       CDTURLSessionInterceptor, skipping",obj);
+        }
+    }
```

Comment:
Interceptors shouldn't be called until the request is above to be made, i.e., when `resume` is called.

Also, would this be clearer written:

```java
for (NSObject *obj in self.requestInterceptors) {

    if(![obj conformsToProtocol:@protocol(CDTHTTPInterceptor)]) {
        CDTLogWarn(CDTREPLICATION_LOG_CONTEXT, @"%@ doesn't conform to protocol \
                   CDTURLSessionInterceptor, skipping", obj);
        continue;
    }

    if (![interceptor respondsToSelector:@selector(interceptRequestInContext:)]) {
        CDTLogWarn(CDTREPLICATION_LOG_CONTEXT, @"%@ doesn't respond to \
                   interceptRequestInContext:, skipping", interceptor);
        continue;
    }

    NSObject<CDTHTTPInterceptor> *interceptor =
            (NSObject<CDTHTTPInterceptor> *)obj;
    ctx = [interceptor interceptRequestInContext:ctx];

}
```

Removing the protocol check and assuming makes this simpler:

```java
for (NSObject<CDTHTTPInterceptor> *obj in self.requestInterceptors) {
    if ([obj respondsToSelector:@selector(interceptRequestInContext:)]) {
        ctx = [obj interceptRequestInContext:ctx];
    } else {
        CDTLogWarn(CDTREPLICATION_LOG_CONTEXT, @"%@ doesn't respond to \
                   interceptRequestInContext:, skipping", obj);
    }
}
```